models/user.go

```diff
@@ -354,21 +354,38 @@ func FindUserByEmailChangeToken(tx *storage.Connection, token string) (*User, er
 }
 
 // FindUserWithRefreshToken finds a user from the provided refresh token.
-func FindUserWithRefreshToken(tx *storage.Connection, token string) (*User, *RefreshToken, error) {
+func FindUserWithRefreshToken(tx *storage.Connection, token string) (*User, *RefreshToken, *Session, error) {
 	refreshToken := &RefreshToken{}
 	if err := tx.Where("token = ?", token).First(refreshToken); err != nil {
 		if errors.Cause(err) == sql.ErrNoRows {
-			return nil, nil, RefreshTokenNotFoundError{}
+			return nil, nil, nil, RefreshTokenNotFoundError{}
 		}
-		return nil, nil, errors.Wrap(err, "error finding refresh token")
+		return nil, nil, nil, errors.Wrap(err, "error finding refresh token")
 	}
 
-	user, err := findUser(tx, "id = ?", refreshToken.UserID)
+	user, err := FindUserByID(tx, refreshToken.UserID)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
-	return user, refreshToken, nil
+	var session *Session
+
+	if refreshToken.SessionId != nil {
+		sessionId := *refreshToken.SessionId
+
+		if sessionId != uuid.Nil {
```

Comment:
hmm will sessionId ever be `uuid.Nil` though? i guess this is just a safeguard but isn't it better if we return an explicit error in the event the sessionId is `uuid.Nil`? 

```suggestion
	if refreshToken.SessionId != nil {
		sessionId := *refreshToken.SessionId
```

Reply:
I'd rather avoid returning any errors and just treat `uuid.Nil` as a nil value, otherwise we're going to have edge-case outages all the time.